examples/medplum-provider/src/pages/encounter/EncounterModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { useNavigate, useLocation } from 'react-router-dom';
+import { Button, Select, Modal, Text, Card, Stack } from '@mantine/core';
+import { useState } from 'react';
+import { CodeInput, CodingInput, ResourceInput, useMedplum, ValueSetAutocomplete } from '@medplum/react';
+import { showNotification } from '@mantine/notifications';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { createReference, normalizeErrorString } from '@medplum/core';
+import { Coding, Encounter, ValueSetExpansionContains } from '@medplum/fhirtypes';
+import { usePatient } from '../../hooks/usePatient';
+
+export const EncounterModal = (): JSX.Element => {
+  const navigate = useNavigate();
+  const medplum = useMedplum();
+  const location = useLocation();
+  const patient = usePatient();
+  const [types, setTypes] = useState<ValueSetExpansionContains[]>([]);
+  const [encounterClass, setEncounterClass] = useState<Coding | undefined>();
+  // Todo: create a resusable type
+  const [status, setStatus] = useState<
+    | 'planned'
+    | 'arrived'
+    | 'triaged'
+    | 'in-progress'
+    | 'onleave'
+    | 'finished'
+    | 'cancelled'
+    | 'entered-in-error'
+    | 'unknown'
+  >('planned');
+  const isOpen = location.pathname.endsWith('/Encounter/new');
+
+  const handleClose = (): void => {
+    navigate(-1);
+  };
+
+  const handleCreateEncounter = (): void => {
+    if (!patient || !encounterClass) {
+      return;
+    }
+
+    const encounterData: Encounter = {
+      resourceType: 'Encounter',
+      status: status,
+      statusHistory: [],
+      class: encounterClass,
+      classHistory: [],
+      type: [
+        {
+          coding: types,
+        },
+      ],
+      subject: createReference(patient),
+    };
+
+    medplum
+      .createResource(encounterData)
+      .then((encounter) => {
+        showNotification({
+          icon: <IconCircleCheck />,
+          title: 'Success',
+          message: 'Encounter created',
+        });
+        console.log('Encounter created:', encounter);
+        // Redirect to actual chart
+        navigate(`/Patient/6d8f3e99-f763-422c-946d-68c95e394a82/Encounter/chart`);
```

Comment:
1. I think this should be inside the `then()` block
2. Need to use the actual Patient, not hardcoded string

```suggestion
        navigate(`/Patient/${patient.id}/Encounter/chart`);
```